src/pages/user/MyWebinars/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useContext, useEffect, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import { useHistory } from "react-router-dom";
+import ProfileLayout from "@/components/Profile/ProfileLayout";
+import { EscolaLMSContext } from "@escolalms/sdk/lib/react/context";
+import { Tabs } from "@escolalms/components/lib/components/atoms/Tabs/Tabs";
+import ProfileWebinars from "@/components/Profile/ProfileWebinars";
+
+export enum WebinarStatus {
+  UPCOMING = "in_coming",
+  PAST = "is_ended",
+  STARTED = "is_started",
+}
+
+const MyWebinarsStyled = styled.section`
+  margin-top: -70px;
+  .tabs-menu {
+    margin: 0 0 70px 40px;
+  }
+`;
+
+const MyWebinarsPage = () => {
+  const { userWebinars, fetchUserWebinars } = useContext(EscolaLMSContext);
+  const { user } = useContext(EscolaLMSContext);
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    fetchUserWebinars();
+  }, [fetchUserWebinars]);
+
+  useEffect(() => {
+    if (!user.loading && !user.value) {
+      history.push("/login");
+    }
```

Comment:
Isn't it better to just protect this route?

Reply:
I mean use some kind of pattern to protect it (instead of per view useEffect etc).